### PR TITLE
Split eventsourced remember entity updates into batched StateChange events based on `max-updates-per-write` config

### LINKED ISSUE #32475: `EventSourcedRememberEntitiesShardStore` not honouring `max-updates-per-write` setting
Upon inspection of the [`RememberEntitiesShardStore.Update` handler in `EventSourcedRememberEntitiesShardStore`](https://github.com/akka/akka/blob/aaf7b8748dbd0622db0dbee9c91e1f646497c8e5/akka-cluster-sharding/src/main/scala/akka/cluster/sharding/internal/EventSourcedRememberEntitiesShardStore.scala#L94) it appears that the code is not doing what was originally intended in #29233.
It appears the `events` list will only ever contain 1 `EntitiesStarted` and/or 1 `EntitiesRemoved` each with the ids of all the entities started or removed, irrespective of their count. 
The later code that was intended to split the entities across multiple writes thus takes no notice of the number of entities in each of those events and thus `max-updates-per-write` will not have any effect.
This means with a large number of simultaneous entity activations or deactivations the journal could receive huge events to persist.

### PR DESCRIPTION
**Pull request by nathanmbrown**
Resolves #32475

### CODE REVIEW

**Comment by lightbend-cla-validator** [comment]
At least one commit author (nbrown@ureason.com) is not linked to a user. See https://help.github.com/en/articles/why-are-my-commits-linked-to-the-wrong-user#commits-are-not-linked-to-any-user

**Comment by lightbend-cla-validator** [comment]
At least one commit author (nbrown@ureason.com) is not linked to a user. See https://help.github.com/en/articles/why-are-my-commits-linked-to-the-wrong-user#commits-are-not-linked-to-any-user

**Comment by lightbend-cla-validator** [comment]
Hi @nathanmbrown,

Thank you for your contribution! We really value the time you've taken to put this together.

We see that you have signed the Lightbend Contributors License Agreement before, however, the CLA has changed since you last signed it.
Please review the new CLA and sign it before we proceed with reviewing this pull request:

[http://www.lightbend.com/contribute/cla](http://www.lightbend.com/contribute/cla)

**Comment by nathanmbrown** [comment]
CLA signed. The original commit had an incorrect old email address on it (nbrown@ureason.com), I have fixed that by pushing a corrected commit.

**Review by patriknw — APPROVED** [review]
LGTM, good catch and thanks for fixing

**Review by johanandren — APPROVED** [review]
LGTM

File: akka-cluster-sharding/src/main/scala/akka/cluster/sharding/internal/EventSourcedRememberEntitiesShardStore.scala [hunk 1]
```diff
@@ -93,30 +106,21 @@ private[akka] final class EventSourcedRememberEntitiesShardStore(
 
     case RememberEntitiesShardStore.Update(started, stopped) =>
       val events =
-        (if (started.nonEmpty) EntitiesStarted(started) :: Nil else Nil) :::
-        (if (stopped.nonEmpty) EntitiesStopped(stopped) :: Nil else Nil)
+        (if (started.nonEmpty) createEntityEvents(started, EntitiesStarted, maxUpdatesPerWrite) else Nil) :::
+        (if (stopped.nonEmpty) createEntityEvents(stopped, EntitiesStopped, maxUpdatesPerWrite) else Nil)
```
Comment (johanandren) [inline akka-cluster-sharding/src/main/scala/akka/cluster/sharding/internal/EventSourcedRememberEntitiesShardStore.scala:110]: Oh, one small thing for Scala 3 compatibility:
```suggestion
        (if (started.nonEmpty) createEntityEvents(started, EntitiesStarted.apply _, maxUpdatesPerWrite) else Nil) :::
        (if (stopped.nonEmpty) createEntityEvents(stopped, EntitiesStopped.apply _, maxUpdatesPerWrite) else Nil)
```

Comment (nathanmbrown) [reply akka-cluster-sharding/src/main/scala/akka/cluster/sharding/internal/EventSourcedRememberEntitiesShardStore.scala:110]: Thanks, I was wondering why the Scala 3 compatibility wasn’t passing!

**Comment by johanandren** [comment]
Sorry, one more place with the same thing, in the test

**Pull request merged**